interp/interp.go

```diff
@@ -159,7 +159,11 @@ func (pc PiecewiseConstant) Predict(x float64) float64 {
 // is assumed to be +Inf. If no such i is found, it returns -1. It assumes that len(xs) >= 2
```

Comment:
The comment here is incorrect (has always been). I think we need to figure out whether the code is correct or the comment is.

Reply:
I've removed the erroneous comment and added tests for the edge cases to verify that the code is working as described in the comment.